### PR TITLE
add copy methods for pulp_deb

### PR DESCRIPTION
**Pull request by aschieb**
Implemented importing for Debian Packages.
Adds support for copying Packages from one Debian Repository to another.

### CODE REVIEW

**Comment by mhrivnak** [comment]
ok test

**Comment by dkliban** [comment]
ok test

**Comment by rbarlow** [comment]
ok test

File: extensions_admin/pulp_deb/extensions/admin/cudl.py [hunk 1]
```diff
@@ -195,3 +196,26 @@ def _all_repos(self, query_params, **kwargs):
...
+        :param type_id: the type id for which we need to get the formatter
+        :type type_id: str
+        :returns: a function to provide a user readable formatted name for a type
+        :rtype: function
+        """
+        return _get_details
+
+
+def _get_details(package):
+    return '%s-%s-%s' % (package['name'], package['version'], package['architecture'])
```
Comment (rbarlow) [inline extensions_admin/pulp_deb/extensions/admin/cudl.py]: Why not just put this return line in the get_formatter_for_type() method?

File: run-tests.py [hunk 2]
```diff
@@ -23,5 +23,4 @@
 dir_safe_all_platforms = [os.path.join(os.path.dirname(__file__), x) for x in TESTS]
 dir_safe_non_rhel5 = [os.path.join(os.path.dirname(__file__), x) for x in PLUGIN_TESTS]
 
-sys.exit(run_tests(PACKAGES, dir_safe_all_platforms, dir_safe_non_rhel5,
-                   flake8_paths=[PROJECT_DIR]))
+sys.exit(run_tests(PACKAGES, dir_safe_all_platforms, dir_safe_non_rhel5))
```
Comment (rbarlow) [inline run-tests.py]: Why remove the flake8 test?

File: extensions_admin/pulp_deb/extensions/admin/cudl.py [hunk 3]
```diff
@@ -195,3 +196,26 @@ def _all_repos(self, query_params, **kwargs):
...
+    def get_formatter_for_type(self, type_id):
+        """
+        Hook to get a the formatter for a given type
+
+        :param type_id: the type id for which we need to get the formatter
+        :type type_id: str
+        :returns: a function to provide a user readable formatted name for a type
+        :rtype: function
+        """
+        return _get_details
```
Comment (rbarlow) [inline extensions_admin/pulp_deb/extensions/admin/cudl.py]: Please add a unit test for this method.

File: extensions_admin/pulp_deb/extensions/admin/pulp_cli.py [hunk 4]
```diff
@@ -50,6 +51,7 @@ def add_repo_section(context, parent_section):
     repo_section.add_command(UpdateDebRepositoryCommand(context))
     repo_section.add_command(cudl.DeleteRepositoryCommand(context))
     repo_section.add_command(ListDebRepositoriesCommand(context))
+    repo_section.add_command(CopyDebUnitCommand(context))
```
Comment (rbarlow) [inline extensions_admin/pulp_deb/extensions/admin/pulp_cli.py]: You should add an assertion for this line in the unit tests here:

https://github.com/pulp/pulp_deb/blob/master/extensions_admin/test/unit/admin/test_pulp_cli.py#L33

File: plugins/pulp_deb/plugins/importers/web.py [hunk 5]
```diff
@@ -98,6 +98,20 @@ def sync_repo(self, repo, sync_conduit, config):
...
+            repo = Repository.objects.get(repo_id=import_conduit.source_repo_id)
+            units = find_repo_content_units(repo, yield_content_unit=True)
+
+        # Associate to the new repository
+        units_to_return = []
+        for u in units:
+            units_to_return.append(u)
+            import_conduit.associate_unit(u)
+
+        return units_to_return
```
Comment (rbarlow) [inline plugins/pulp_deb/plugins/importers/web.py]: Please add a few unit tests for this method. Here are some suggestions:

0) Test when units is None.
1) Test when units is not None.
2) Test when units is [].

**Comment by rbarlow** [comment]
Nice work so far! A few notes:

0) Add some unit tests. If you need guidance, let me know and I'll give you some more specific pointers.
1) Let's not drop the flake8 test.
2) I think that small refactor of moving the return statement to the other method is worth it, as it reduces the depth of the stack and cuts the methods that need unit testing in half.

Let me know when to take another look!

**Comment by aschieb** [comment]
I don't know exactly why I removed the flake8 test, I was surprised I did that.
Now the test is back in and all flake8 issues are fixed.
Would you like to review my unit tests?
I wrote them like the ones in the RPM ISO importer.

**Comment by mmoll** [comment]
@rbarlow what's the status here? :)

**Comment by rbarlow** [comment]
Thank you for your contribution! I apologize for my delay in re-reviewing. Looks good to me!

**Pull request merged**